### PR TITLE
Improvement/os information gathering (#24)

### PR DESCRIPTION
**Pull request by stratus-ss**
This PR adds the ability for the program to first look for a VMWare tools column and if that row is empty, it will then grab the value from the VM Config column.

In addition, in order to support better header handling, this PR also changes the set_column_headings() function so that it now identifies which version of headers is likely the best match (using LOGGER.debug) and then determining which headers from the closest version are missing and provides this via text output

### CODE REVIEW

**Review by csfreak — CHANGES_REQUESTED** [review]
beyond the two changes above,  I dont like the changes to set_column_headings, but I can't think of a better way to do it, and I dont inherently disagree with the value It provides, so I will not object to it.

File: vminfo_parser/vminfo_parser.py [hunk 1]
```diff
@@ -58,29 +58,54 @@ def from_file(cls: t.Self, filepath: Path) -> "VMData":
...
+
+        if best_match is None:
+            raise ValueError("No matching header set found")
+
+        self.column_headers = const.COLUMN_HEADERS[best_match].copy()
+        missing_headers = [header for header in self.column_headers.values() if header not in self.df.columns]
+        self.column_headers["unitType"] = "GB" if best_match == "VERSION_1" else "MB"
+
+        if missing_headers:
+            LOGGER.debug(f"Using VERSION_{best_match} as the closest match.")
```
[the remark this answers is not in the document]
Comment (stratus-ss) [reply vminfo_parser/vminfo_parser.py]: moved outside of if block as suggested

File: vminfo_parser/vminfo_parser.py [hunk 2]
```diff
@@ -58,29 +58,54 @@ def from_file(cls: t.Self, filepath: Path) -> "VMData":
...
+            LOGGER.critical("The following headers are missing:")
+            for header in missing_headers:
+                LOGGER.critical(f"- {header}")
```
[the remark this answers is not in the document]
Comment (stratus-ss) [reply vminfo_parser/vminfo_parser.py]: suggestion taken, for-loop provided form but not function so it was removed

**Review by csfreak — APPROVED** [review]
lgtm

**Pull request merged**